### ISSUE TITLE
Unable to rollout the operator

### ISSUE THREAD

**Issue by timbrd** [stashed/stash#1007]
I am not able to deploy the operator v0.9.0-rc.2 on OKD 3.11.

While initializing the operator, I always get the following error message:

```
I0105 00:47:11.850796       1 run.go:26] Starting operator version v0.9.0-rc.2+8ce4ab865ccb7b57fc4d98a65669cc7fef8a3c9e ...
I0105 00:47:12.663345       1 lib.go:112] Kubernetes version: &version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0+d4cacc0", GitCommit:"d4cacc0", GitTreeState:"clean", BuildDate:"2019-12-19T05:39:29Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
Error: CustomResourceDefinition.apiextensions.k8s.io "restics.stash.appscode.com" is invalid: spec.validation.openAPIV3Schema: Invalid value: apiextensions.JSONSchemaProps{ID:"", Schema:"", Ref:(*string)(nil), Description:"", Type:"object", Format:"", Title:"", Default:(*apiextensions.JSON)(nil), Maximum:(*float64)(nil), ExclusiveMaximum:false, Minimum:(*float64)(nil), ExclusiveMinimum:false, MaxLength:(*int64)(nil), MinLength:(*int64)(nil), Pattern:"", MaxItems:(*int64)(nil), MinItems:(*int64)(nil), UniqueItems:false, MultipleOf:(*float64)(nil), Enum:[]apiextensions.JSON(nil), MaxProperties:(*int64)(nil), MinProperties:(*int64)(nil), Required:[]string(nil), Items:(*apiextensions.JSONSchemaPropsOrArray)(nil), AllOf:[]apiextensions.JSONSchemaProps(nil), OneOf:[]apiextensions.JSONSchemaProps(nil), AnyOf:[]apiextensions.JSONSchemaProps(nil), Not:(*apiextensions.JSONSchemaProps)(nil), Properties:map[string]apiextensions.JSONSchemaProps{"kind":apiextensions.JSONSchemaProps{ID:"", Schema:"", Ref:(*string)(nil), Description:"Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds ", Type:"string", Format:"", Title:"", Default:(*apiextensions.JSON)(nil), Maximum:(*float64)(nil), ExclusiveMaximum:false, Minimum:(*float64)(nil), ExclusiveMinimum:false, MaxLength:(*int64)(nil), MinLength:(*int64)(nil), Pattern:"", MaxItems:(*int64)(nil), MinItems:(*int64)(nil), UniqueItems:false, MultipleOf:(*float64)(nil), Enum:[]apiextensions.JSON(nil), MaxProperties:(*int64)(nil), MinProperties:(*int64)(nil), Required:[]string(nil), Items:(*apiextensions.JSONSchemaPropsOrArray)(nil), AllOf:[]apiextensions.JSONSchemaProps(nil), OneOf:[]apiextensions.JSONSchemaProps(nil), AnyOf:[]apiextensions.JSONSchemaProps(nil), Not:(*apiextensions.JSONSchemaProps)(nil), Properties:map[string]apiextensions.JSONSchemaProps(nil), AdditionalProperties:(*apiextensions.JSONSchemaPropsOrBool)(nil), PatternProperties:map[string]apiextensions.JSONSchemaProps(nil), Dependencies:apiextensions.JSONSchemaDependencies(nil), AdditionalItems:(*apiextensions.JSONSchemaPropsOrBool)(nil), Definitions:apiextensions.JSONSchemaDefinitions(nil), ExternalDocs:(*apiextensions.ExternalDocumentation)(nil), Example:(*apiextensions.JSON)(nil)}, 
[...]
Definitions:apiextensions.JSONSchemaDefinitions(nil), ExternalDocs:(*apiextensions.ExternalDocumentation)(nil), Example:(*apiextensions.JSON)(nil)}: must only have "properties", "required" or "description" at the root if the status subresource is enabled
```

Installing version v0.9.0-rc.1 works flawlessly. I had to disable the cr subresource status checking though (--enable-status-subresource=false), which [has been removed](https://github.com/stashed/installer/pull/15) in rc1.

Any advice on this? 
Has anyone tested version 0.9.0-rc2 on OpenShift 3.11?

**Comment by tamalsaha** [comment]
What is your matching Kubernetes version ? 1.11 ?

**Comment by timbrd** [comment]
> What is your matching Kubernetes version ? 1.11 ? Stash requires Kubenetes 1.12 or later version.

Yes. The exact version is in the logmessage above: 
```
Kubernetes version: &version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0+d4cacc0", [...]
```

**Comment by tamalsaha** [comment]
This seems like a known issue for Kubernetes 1.11 release. https://github.com/jetstack/cert-manager/issues/2200 . This issue has more details on this.

**Comment by tamalsaha** [comment]
@timbrd , you can try the v0.9.0-rc.1 release. That might work.

I have opened pr #1008 that should fix this issue on master. We are looking to cut a new release soon (probably next week). That should include this fix.

**Comment by timbrd** [comment]
> I have opened pr #1008 that should fix this issue on master. We are looking to cut a new release soon (probably next week). That should include this fix.

Great to hear!

I will then use the last release in the meantime as suggested.

**Comment by timbrd** [comment]
@tamalsaha Did you find a way to run the tests against an openshift cluster?

**Comment by timbrd** [comment]
Are there still plans to support OpenShift 3.11? I think you guys have put a lot of effort into being able to test stash against Kubernetes 1.11.

**Comment by tamalsaha** [comment]
@timbrd , sorry for the slow response. I have not been able to get https://github.com/openshift/origin/issues/24377 working . I did not have time to look into this in the last week.

I also looked into if there is a way to support OpenShift 4.x. But it seemed to me that they don't have a similar Docker based installer approach. 

I was able to get rancher folks to cut a new release with the fixes needed for k8s 1.11 https://github.com/rancher/local-path-provisioner/releases/tag/v0.0.12 .

I would like to support 3.11 but without support in CI system I don't know how to make it happen. And given my all other responsibilities, I have hard time further debugging the issue https://github.com/openshift/origin/issues/24377 myself. If you can get this issue fixed, I am happy to get the fixes necessary for Stash.

If you are doing this on behalf of a company and they would rather hire us as consultants to get this looked into further in an expedited fashion, please send me an email tamal@appscode.com . I can give you a quote for our hourly rates.

**Comment by timbrd** [comment]
@tamalsaha  Do you really need rancher's local path provisioner for your tests? `oc cluster up` seems to have a few preconfigured pvs and even a dummy dynamic provisioner:

```
$ oc cluster up
Getting a Docker client ...
Checking if image openshift/origin-control-plane:v3.11 is available ...
Checking type of volume mount ...
Determining server IP ...
[...]

$ oc get storageclass
No resources found.

$ oc get pv | head
oc get pv | head
NAME      CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM                     STORAGECLASS   REASON    AGE
pv0001    100Gi      RWO,ROX,RWX    Recycle          Available                                                      21m
pv0002    100Gi      RWO,ROX,RWX    Recycle          Available                                                      21m
pv0003    100Gi      RWO,ROX,RWX    Recycle          Available                                                      21m
pv0004    100Gi      RWO,ROX,RWX    Recycle          Available                                                      21m
pv0005    100Gi      RWO,ROX,RWX    Recycle          Available                                                      21m
pv0006    100Gi      RWO,ROX,RWX    Recycle          Available                                                      21m
pv0007    100Gi      RWO,ROX,RWX    Recycle          Available                                                      21m
pv0008    100Gi      RWO,ROX,RWX    Recycle          Available                                                      21m
pv0009    100Gi      RWO,ROX,RWX    Recycle          Available                                                      21m

$ oc create -f - <<EOF
> kind: PersistentVolumeClaim
> apiVersion: v1
> metadata:
>     name: test
> spec:
>     accessModes:
>       - ReadWriteOnce
>     resources:
>       requests:
>         storage: 1Gi
> EOF
persistentvolumeclaim/test created

$ oc get pvc
NAME      STATUS    VOLUME    CAPACITY   ACCESS MODES   STORAGECLASS   AGE
test      Bound     pv0098    100Gi      RWO,ROX,RWX                   24s
```

**Comment by tamalsaha** [comment]
If you run 

`kubectl apply -f https://raw.githubusercontent.com/tamalsaha/okd-github-actions-demo/master/hack/kubernetes/pod.yaml`

Does does pod run?

**Comment by tamalsaha** [comment]
So, I just tried it locally. As long as, we don't define a storageclass in the PVC, it works in OpenShift 3.11 .

Works: https://github.com/tamalsaha/okd-github-actions-demo/blob/master/hack/kubernetes/pod-no-storageclass.yaml

Does not work: https://raw.githubusercontent.com/tamalsaha/okd-github-actions-demo/master/hack/kubernetes/pod.yaml

This is weird. But we can add some logic to not specify any storageclass for testing against OpenShift. That might work.

**Comment by timbrd** [comment]
Yes, I though it was possible to create a storageclass first, but I don't know which provisionier to use. `provisioner: kubernetes.io/no-provisioner` unfortunately does not work, the pvc is stuck in pending state forever.

It would be great if you customize the tests for OpenShift 3.

**Comment by tamalsaha** [comment]
We are very close to cutting v0.9.0-rc.3 for Stash. @hossainemruz is finishing up last minute doc updates. We can take a look at this after this release is out.

**Comment by timbrd** [comment]
@tamalsaha Are there any news on the issue?

**Comment by hossainemruz** [comment]
@timbrd We have released `v0.9.0-rc.6`. We have brought back Kubernetes 1.11.x support. We have also removed defaulting `fsGroup` which was causing problem in Openshift. Ref: #1009.

New release should work. Can you please try and let us know?

**Comment by hossainemruz** [comment]
Closing for inactivity
